compiler/il/OMRBlock.cpp

```diff
@@ -1334,23 +1334,27 @@ OMR::Block::splitEdge(TR::Block *from, TR::Block *to, TR::Compilation *c, TR::Tr
       insertionExit->join(newBlock->getEntry());
       newBlock->getExit()->join(insertionEntry);
 
-      if (isExceptionEdge)
+      //OSRCatchBlock is handled differently from normal catch block
+      //the control flow is from -> exceptionEdge -> newBlock -> normalEdge -> to
+      if (isExceptionEdge && to->isOSRCatchBlock())
          {
-         if (to->isOSRCatchBlock())
-            newBlock->setHandlerInfoWithOutBCInfo(
-                  TR::Block::CanCatchOSR, 
-                  to->getInlineDepth(), 
-                  -1, 
-                  to->getOwningMethod(), 
-                  c);
-         else
-            newBlock->setHandlerInfo(
-               to->getCatchType(),
-               to->getInlineDepth(),
-               to->getHandlerIndex(),
-               to->getOwningMethod(),
+         newBlock->setHandlerInfoWithOutBCInfo(
```

Comment:
I am worried about the BCInfo of the catch since the BC info is going to match the induction point and probably not the BCinfo of the block. It, in general, should not make a difference, but the potential difference makes me worry about a hard to find bug possibility resulting from this code.